lib/auth/auth_with_roles.go

```diff
@@ -4753,6 +4753,9 @@ func (a *ServerWithRoles) AcquireSemaphore(ctx context.Context, params types.Acq
 	if err := a.action(apidefaults.Namespace, types.KindSemaphore, types.VerbCreate, types.VerbUpdate); err != nil {
 		return nil, trace.Wrap(err)
 	}
+	if a.hasBuiltinRole(types.RoleDiscovery) && params.SemaphoreKind != types.KindAccessGraph {
+		return nil, trace.AccessDenied("discovery service can not create other semaphores")
```

Comment:
It's not clear "other than what" semaphores, we should say other than what type or just say that this kind is forbidden.

```suggestion
		return nil, trace.AccessDenied("discovery service can not create semaphore of kind %q", params.SemaphoreKind)
```

Reply:
I don't want to leak the semaphore kind. Mainly because if we run `AcquireSemaphoreLock` the error is completely hidden.